### PR TITLE
fix(videos):play/pause videos on visibility 

### PR DESCRIPTION
**Pull request by vishu2222**
fixes: https://github.com/quintype/page-builder/issues/2341

### CODE REVIEW

File: src/components/story-elements/brightcove.js [hunk 1]
```diff
@@ -68,12 +90,27 @@ const BrightcoveElement = (props) => {
...
-              <img className="brightcove-poster" onClick={() => loadLibrary()} src={posterImage} alt="video" />
+              <img
+                className="brightcove-poster"
+                onClick={() => {
+                  loadLibrary();
+                  setThumbnailClicked(true);
+                }}
+                src={posterImage}
+                alt="video"
+              />
```
Comment (nandap4790) [inline src/components/story-elements/brightcove.js:113]: the onclick can be added to the container directly

**Review by nandap4790 — CHANGES_REQUESTED** [review]

File: src/components/story-elements/brightcove.js (hunk 1 above)
Comment (vishu2222) [reply src/components/story-elements/brightcove.js]: removed onclick event repeatition

File: src/components/story-elements/brightcove.js [hunk 2]
```diff
@@ -68,22 +90,31 @@ const BrightcoveElement = (props) => {
...
-      <div className="brightcove-wrapper">
+      <div className="brightcove-wrapper" ref={videoRef}>
         {!showVideo && (
-          <>
-            <button className="brightcove-playBtn" onClick={() => loadLibrary()} aria-label="Play Video" />
+          <div
+            onClick={() => {
+              loadLibrary();
+              setThumbnailClicked(true);
+            }}
```
Comment (ags1773) [inline src/components/story-elements/brightcove.js:99]: change to async

**Review by nandap4790 — APPROVED** [review]

File: src/components/story-elements/brightcove.js (hunk 2 above)
Comment (vishu2222) [reply src/components/story-elements/brightcove.js]: done

**Review by ags1773 — APPROVED** [review]

**Comment by ags1773** [comment]
Please add some comments describing the intersection observers and play/pause behaviour

**Review by ags1773 — APPROVED** [review]

**Review by nandap4790 — APPROVED** [review]

**Pull request merged**